client/state/index.js

```diff
@@ -111,7 +114,40 @@ export const reducer = combineReducers( {
 	ui,
 	users,
 	wordads,
-} );
+};
+
+let reducers = { ...initialReducers };
+let onChange = null;
+
+// TODO: See if this export can be eliminated.
+export let reducer = combineReducers( reducers );
+
+export function addReducer( name, reducerFunc ) {
+	if ( reducerExists( name ) ) {
+		throw new Error( 'addReducer(): name "' + name + '" already in use' );
+	}
+
+	reducers = { ...reducers, [ name ]: reducerFunc };
+
+	if ( onChange ) {
+		onChange();
+	}
+}
+
+export function removeReducer( name ) {
+	const { [ name ]: removedReducer, ...remainingReducers } = reducers;
+
+	if ( ! removedReducer ) {
+		log( 'removeReducer(): name not found' );
+		return;
+	}
+
+	reducers = remainingReducers;
+}
+
+export function reducerExists( name ) {
+	return ( reducers[ name ] !== undefined );
+}
```

Comment:
what might use this outside of the module itself?

if this is the only module needing to use this function then we could easily allow it to accept the reducer object…

```js
export const reducerExists = ( reducers, name ) => reducers[ name ] !== undefined;
```

of course at this point we recognize a common code pattern we might be able to reuse…

```js
if ( has( reducers, name ) ) {
	…
}
```